### PR TITLE
Allow specifying custom implementation of k-rpc

### PR DESCRIPTION
**Pull request by nazar-pc**
I'm working on https://github.com/webtorrent/webtorrent/issues/288 and similarly to https://github.com/mafintosh/k-rpc/pull/8 I need to be able to provide a custom implementation of k-rpc.

### CODE REVIEW

File: client.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function DHT (opts) {
   this._peers = new PeerStore(opts.maxPeers || 10000)
 
   this._secrets = null
-  this._rpc = krpc(opts)
+  this._rpc = opts.krpc || krpc(opts)
```
Comment (JakeChampion) [inline client.js]: Should this be documented in the README?

Comment (nazar-pc) [inline client.js]: Updated readme here and in k-rpc

**Comment by RationalCoding** [comment]
Why not just use `npm shrinkwrap` to override the nested dependency?

https://docs.npmjs.com/cli/shrinkwrap

```javascript
{
    "name": "webtorrent-dht",
    "dependencies": {
        "bittorrent-dht": {
            "version": "0.0.1",
            "dependencies": {
                "k-rpc": {  "version": "git@github.com:..." } <- Your custom k-rpc repo
            }
        }
    }
}
```

**Comment by nazar-pc** [comment]
1) It is very verbose
2) It is ugly
3) It is a workaround rather than proper solution

Besides everything above making it an argument does not require sophisticated tooling and is much easier to test. Also when/if the name of dependency changes, the name of the argument might be kept the same.

I hope it wouldn't take years to merge this single line PR.

**Comment by RationalCoding** [comment]
Yes, I agree it's less aesthetic. I assumed you wouldn't want to bundle two versions of the same module when you Browserify webtorrent-dht.

**Comment by nazar-pc** [comment]
I don't need to just replace k-rpc, replacement itself inherits k-rpc, so I will have k-rpc bundled anyway. Making it like you've suggested will, probably, work, but it is very tricky approach:)

**Comment by nazar-pc** [comment]
Can someone review and merge this PR, please?

**Comment by feross** [comment]
Released as 7.7.0

**Pull request merged**